### PR TITLE
decoder: Fix memleak because of shadowed variable

### PR DESCRIPTION
**Pull request by pent0**

### CODE REVIEW

File: vita3k/codec/src/mp3.cpp [hunk 1]
```diff
@@ -170,3 +170,10 @@ Mp3DecoderState::Mp3DecoderState(uint32_t channels) {
     int err = avcodec_open2(context, codec, nullptr);
     assert(err == 0);
 }
+
+Mp3DecoderState::~Mp3DecoderState() {
+    avcodec_close(context);
+    av_free(context);
+    
+    context = nullptr;
+}
```
Comment (Zangetsu38) [inline vita3k/codec/src/mp3.cpp:179]: Add new line

File: vita3k/ngs/src/modules/atrac9.cpp [hunk 2]
```diff
@@ -53,7 +53,7 @@ void Module::process(KernelState &kern, const MemState &mem, const SceUID thread
     State *state = data.get_state<State>();
 
     assert(state);
-
+
```
Comment (Zangetsu38) [inline vita3k/ngs/src/modules/atrac9.cpp:56]: revert this file ^^

Comment (pent0) [reply vita3k/ngs/src/modules/atrac9.cpp:56]: yes format tell me

**Review by Zangetsu38 — APPROVED** [review]

**Review by Zangetsu38 — APPROVED** [review]

File: vita3k/codec/src/atrac9.cpp [hunk 3]
```diff
@@ -182,3 +182,10 @@ Atrac9DecoderState::Atrac9DecoderState(uint32_t config_data)
     int err = avcodec_open2(context, codec, nullptr);
     assert(err == 0);
 }
+
+Atrac9DecoderState::~Atrac9DecoderState() {
```
[the remark this answers is not in the document]
Comment (pent0) [reply vita3k/codec/src/atrac9.cpp:186]: Yes, avdec_free_context for some reason only works for video

**Pull request merged**